discovery-handlers/opcua/src/discovery_handler.rs

```diff
@@ -121,14 +138,19 @@ impl DiscoveryHandler for DiscoveryHandlerImpl {
                         );
                         properties
                             .insert(OPCUA_DISCOVERY_URL_LABEL.to_string(), discovery_url.clone());
-                        Device {
-                            id: discovery_url,
+                        let query_body = QueryDevicePostBody{
+                            discovery_url:discovery_url.clone(),
+                            protocol: "OPC".to_string()
+                        };
+                        DeviceQueryInput{
+                            id:discovery_url,
                             properties,
-                            mounts: Vec::default(),
-                            device_specs: Vec::default(),
+                            query_device_payload:Some(serde_json::to_string(&query_body).unwrap_or(String::from("{}"))),
+                            mounts:Vec::default(),
                         }
                     })
-                    .collect::<Vec<Device>>();
+                    .collect::<Vec<DeviceQueryInput>>();
+                let discovered_devices= query_devices(device_query_requests,query_device_http.clone()).await;
```

Comment:
I may be missing something, but I dont see where we are checking if a `query_device_http` was provided before calling `query_devices`

Reply:
The http endpoint check is actually inside query_devices. You can call query_devices from discovery handler without http query. In that case, query_devices simply assemble Devices without insert external properties, the block from L102 will be skipped.
https://github.com/project-akri/akri/blob/8dc09fdc891659cc1d0ff60da9630ebb5aa1d92b/discovery-utils/src/registration_client.rs#L100-L102